### PR TITLE
Add mode to ProxyDefault configEntry

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:
- Update ProxyDefaults with the field `mode`.

How I've tested this PR: Acceptance tests

How I expect reviewers to test this PR: Code review

Checklist:
- [ ] Tests added
- [ ] CHANGELOG entry added (*HashiCorp engineers only, community PRs should not add a changelog entry*)

### CODE REVIEW

File: api/v1alpha1/proxydefaults_types.go [hunk 1]
```diff
@@ -66,6 +66,14 @@ type ProxyDefaultsSpec struct {
 	// Note: This cannot be set using the CRD and should be set using annotations on the
 	// services that are part of the mesh.
 	TransparentProxy *TransparentProxy `json:"transparentProxy,omitempty"`
+	// Mode can be one of direct or transparent. transparent represents that inbound and outbound
```
Comment (ndhanushkodi) [inline api/v1alpha1/proxydefaults_types.go:69]:
```suggestion
	// Mode can be one of "direct" or "transparent". "transparent" represents that inbound and outbound
```

File: api/v1alpha1/proxydefaults_types.go [hunk 2]
```diff
@@ -66,6 +66,14 @@ type ProxyDefaultsSpec struct {
...
+	// application traffic is being captured and redirected through the proxy. This mode does not
+	// enable the traffic redirection itself. Instead it signals Consul to configure Envoy as if
+	// traffic is already being redirected. direct represents that the proxy's listeners must be
```
Comment (ndhanushkodi) [inline api/v1alpha1/proxydefaults_types.go:72]:
```suggestion
	// enable the traffic redirection itself. Instead it signals Consul to configure Envoy as if
	// traffic is already being redirected. "direct" represents that the proxy's listeners must be
```

File: config/crd/bases/consul.hashicorp.com_proxydefaults.yaml [hunk 3]
```diff
@@ -81,6 +81,9 @@ spec:
                     description: Mode is the mode that should be used for the upstream connection. One of none, local, or remote.
                     type: string
                 type: object
+              mode:
+                description: 'Mode can be one of direct or transparent. transparent represents that inbound and outbound application traffic is being captured and redirected through the proxy. This mode does not enable the traffic redirection itself. Instead it signals Consul to configure Envoy as if traffic is already being redirected. direct represents that the proxy''s listeners must be dialed directly by the local application and other proxies. Note: This cannot be set using the CRD and should be set using annotations on the services that are part of the mesh.'
```
Comment (ndhanushkodi) [inline config/crd/bases/consul.hashicorp.com_proxydefaults.yaml:85]: same suggestion as above in the comment for the field.

**Review by ndhanushkodi — APPROVED** [review]
Looks great Ashwin! Just a nit suggestion.

**Review by kschoche — APPROVED** [review]
Looks great!

**Pull request merged**